src/timer.rs

```diff
@@ -30,6 +30,17 @@ pub enum TimerChannelNum {
     Channel3,
 }
 
+impl From<TimerChannelNum> for usize {
+    fn from(channel: TimerChannelNum) -> Self {
+        match channel {
+            TimerChannelNum::Channel0 => 0,
+            TimerChannelNum::Channel1 => 1,
+            TimerChannelNum::Channel2 => 2,
+            TimerChannelNum::Channel3 => 3,
+        }
+    }
+}
+
```

Comment:
there's a crate num_enum in this repo that can generate this implementation for you - you just need to do

```
#[derive(num_enum::IntoPrimitive, num_enum::TryFromPrimitive)]
#[repr(usize)]
```
and then set the variants to whatever you want the numeric values to be